### PR TITLE
Week2 documentation unittests

### PR DESCRIPTION
**Pull request by foger3**

### CODE REVIEW

File: twitter_nlp/core.py [hunk 1]
```diff
@@ -1,10 +1,69 @@
...
+        for tweet in tweet_tag:
+            for pair in tweet:
+                tag = pair[1]
+                if tag == "NN":
+                    noun_count += 1
+                elif tag == "JJ":
+                    adj_count += 1
+                elif tag == "VB":
+                    verb_count += 1
+        self.counts = [noun_count, adj_count, verb_count]
```
Comment (nebnebb) [inline twitter_nlp/core.py:69]: `self.counts` is not yet known as an instance variable to the `nlp` class. You should consider initializing the counts instance variable in the init (by setting it to None for example).

See this StackOverflow answer for more information:
https://stackoverflow.com/a/19292653

File: twitter_nlp/core.py [hunk 2]
```diff
@@ -1,10 +1,69 @@
...
+
+Todo:
+    * incorporate tweet generator by using twitter API package
+    * incorpoarte full sentiment analysis
+"""
+import nltk
+from nltk.corpus import twitter_samples
+from nltk.tag import pos_tag_sents
+
+class nlp:
```
Comment (nebnebb) [inline twitter_nlp/core.py:23]: Classes in python should be in PascalCase (i.e., starting with a capital letter). In general, abbreviations should be in all caps, so you can consider changing the name of this class to `NLP` (all capital letters),

File: twitter_nlp/core.py [hunk 3]
```diff
@@ -1,10 +1,69 @@
...
+        noun_count = 0
+        adj_count = 0
+        verb_count = 0
+        tweet_token = twitter_samples.tokenized(self.tweets)
+        tweet_tag = pos_tag_sents(tweet_token) 
+
+        for tweet in tweet_tag:
+            for pair in tweet:
+                tag = pair[1]
+                if tag == "NN":
```
Comment (nebnebb) [inline twitter_nlp/core.py:63]: These if-statements compare the tag against so-called "magic strings". Consider declaring the strings `"NN"`, `"JJ"`, `"VB"` as constant variables with a descriptive name.

For more info, see 
https://softwareengineering.stackexchange.com/a/365344

File: twitter_nlp/core.py [hunk 4]
```diff
@@ -1,10 +1,69 @@
...
+
+        Args:
+            tweets (str): String indicating the tweet list to be used.
+                Strings to be choosen from are 'positive_tweets.json',
+                'negative_tweets.json', 'tweets.20150430-223406.json'
+                (default is 'positive_tweets.json').
+        """
+        self.tweets = tweets 
+
+    def simple_nlp(self):
```
Comment (nebnebb) [inline twitter_nlp/core.py:47]: You can consider separating the functionality of reading/loading the twitter_samples data and calculating the different "counts". This way, you can more easily test the "count" functionality by several different inputs via unit tests (and cover edge cases etc).

File: twitter_nlp/test/test_nlp.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    """Testing unit to determine if output remains the same.
+
+    Basic unittest to determine if the outputs from the counting 
+    function (simple_nlp) still match the original results.
+    """
+    def test_compute(self):
+        """Tests whether output of class matches the orignal ones."""
+        obj = nlp()
+        obj.simple_nlp()
+        self.assertAlmostEqual(
```
Comment (nebnebb) [inline twitter_nlp/test/test_nlp.py:18]: What kind of behaviour are you testing with this unit test? A test to check if the functionality does change is not really what unit tests should do. You should try to test for a specific intention/behaviour to verify that the programming is working as intended.

It also does not allow for any improvements or alterations to be made to the software, since you only compare if the output is the same as a previous run.

Also, if a fellow programmer alters your code and this test fails, how would they know what is wrong and how to fix it?
